packages/babel-parser/src/parser/expression.js

```diff
@@ -2110,7 +2110,8 @@ export default class ExpressionParser extends LValParser {
     this.state.inParameters = false;
 
     if (isExpression) {
-      node.body = this.parseMaybeAssignAllowIn();
+      // https://tc39.es/ecma262/#prod-ExpressionBody
+      node.body = this.parseMaybeAssign();
```

Comment:
After the `[In]` parameter tracking refactor, the fix is straightforward: `parseMaybeAssign` parses `AssignmentExpression[?In]`, while `parseMaybeAssignAllowIn` parses `AssignmentExpression[In]`.

According to the spec we should replace `parseMaybeAssignAllowIn` to `parseMaybeAssign`.